src/frontend/src/flows/register.ts

```diff
@@ -59,18 +59,19 @@ const init = (): Promise<LoginResult | null> =>
       try {
         const pendingIdentity = WebAuthnIdentity.create({
           publicKey: creationOptions(),
-        }).catch((error) => {
-          resolve(apiResultToLoginResult({ kind: "authFail", error }));
-          // We can never get here, but TS doesn't understand that
-          return 0 as unknown as WebAuthnIdentity;
         });
         await tick();
         // Kick-start both the captcha creation and the identity
         const captcha = makeCaptcha();
-        const identity = await pendingIdentity;
-        await captcha;
-        const result = await confirmRegister(captcha, identity, alias);
-        resolve(result);
+        pendingIdentity
+          .catch((error) => {
+            resolve(apiResultToLoginResult({ kind: "authFail", error }));
```

Comment:
Thanks for the suggestions! I'm not sure I understood everything though, I think I lack hands on async experience...

> To transform the error, why not `catch(error => throw apiResultToLoginResult({ kind: "authFail", error }))`? That is the equivalent of `map_err(...)` in Rust and is more idiomatic.

But here the behavior is quite different, correct? In your suggestion, the error isn't `catch`ed and will bubble up to the console. I've tried not to modify the original behavior (written by someone else) where the promise always returns a `LoginResult`, be it success or failure (which is why we `catch` in the first place).

> For running the two promises in parallel, why not use: ...

Here, the behavior is different I believe. It would wait until the captcha has been created before showing the confirmation page, whereas now it kickstarts both but only waits for the identity to be ready. I might be wrong, I'm not very fluent with promises!

Reply:
While the behavior is a bit different, I think `Promise.all` is much cleaner, so I followed your suggestion. PTAL!